src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisWorkerService.java

```diff
@@ -0,0 +1,146 @@
+import com.powsybl.network.store.client.PreloadingStrategy;
+import com.powsybl.security.SecurityAnalysis;
+import com.powsybl.security.SecurityAnalysisResult;
+import org.gridsuite.securityanalysis.server.repository.SecurityAnalysisResultRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.http.HttpStatus;
+import org.springframework.messaging.Message;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Consumer;
+import java.util.logging.Level;
+import java.util.stream.Collectors;
+
+/**
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ */
+@Service
+public class SecurityAnalysisWorkerService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SecurityAnalysisWorkerService.class);
+
+    private static final String CATEGORY_BROKER_INPUT = SecurityAnalysisWorkerService.class.getName()
+            + ".input-broker-messages";
+
+    private NetworkStoreService networkStoreService;
+
+    private ActionsService actionsService;
+
+    private SecurityAnalysisResultRepository resultRepository;
+
+    private ObjectMapper objectMapper;
+
+    private SecurityAnalysisConfigService configService;
+
+    private SecurityAnalysisResultPublisherService resultPublisherService;
+
+    public SecurityAnalysisWorkerService(NetworkStoreService networkStoreService, ActionsService actionsService,
+                                         SecurityAnalysisResultRepository resultRepository, ObjectMapper objectMapper,
+                                         SecurityAnalysisConfigService configService, SecurityAnalysisResultPublisherService resultPublisherService) {
+        this.networkStoreService = Objects.requireNonNull(networkStoreService);
+        this.actionsService = Objects.requireNonNull(actionsService);
+        this.resultRepository = Objects.requireNonNull(resultRepository);
+        this.objectMapper = Objects.requireNonNull(objectMapper);
+        this.configService = Objects.requireNonNull(configService);
+        this.resultPublisherService = Objects.requireNonNull(resultPublisherService);
+    }
+
+    private static String sanitizeParam(String param) {
+        return param != null ? param.replaceAll("[\n|\r|\t]", "_") : null;
+    }
+
+    private Mono<Network> getNetwork(UUID networkUuid) {
+        // FIXME to re-implement when network store service will be reactive
+        return Mono.fromCallable(() -> {
+            try {
+                return networkStoreService.getNetwork(networkUuid, PreloadingStrategy.COLLECTION);
+            } catch (PowsyblException e) {
```

Comment:
if there's an exception are sure that is about network not found ?

Reply:
Looking at the code on network store side, yes.